lib/email_processor.rb

```diff
@@ -6,6 +6,13 @@ def initialize(email)
   end
 
   def process
+
+    # Guard clause to prevent ESPs like Sendgrid from posting over and over again
+    # if the email presented is invalid and generates a 500.  Returns a 200
+    # error as discussed on https://sendgrid.com/docs/API_Reference/Webhooks/parse.html
+    # This error happened with invalid email addresses from PureChat
+    return if @email.from[:email].match(/\A[^@\s]+@([^@\s]+\.)+[^@\s]+\z/).blank?
```

Comment:
I was gonna reply that it would require monkey patching.  @CGA1123 where do you propose putting that method?  Just in `email_processor.rb`? That would help with readability but not so much with sharing the method.